packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingCurrencyFilter.ts

```diff
@@ -0,0 +1,49 @@
+import { CurrencyFilter } from '@/object-record/record-filter/types/ObjectRecordQueryFilter';
+
+export const isMatchingCurrencyFilter = ({
+  currencyFilter,
+  value,
+}: {
+  currencyFilter: CurrencyFilter;
+  value: number;
+}) => {
+  if (currencyFilter?.amountMicros !== undefined) {
+    switch (true) {
+      case currencyFilter.amountMicros.eq !== undefined: {
+        return value === currencyFilter.amountMicros.eq;
+      }
+      case currencyFilter.amountMicros.neq !== undefined: {
+        return value !== currencyFilter.amountMicros.neq;
+      }
+      case currencyFilter.amountMicros.gt !== undefined: {
+        return value > currencyFilter.amountMicros.gt;
+      }
+      case currencyFilter.amountMicros.gte !== undefined: {
+        return value >= currencyFilter.amountMicros.gte;
+      }
+      case currencyFilter.amountMicros.lt !== undefined: {
+        return value < currencyFilter.amountMicros.lt;
+      }
+      case currencyFilter.amountMicros.lte !== undefined: {
+        return value <= currencyFilter.amountMicros.lte;
+      }
+      case currencyFilter.amountMicros.in !== undefined: {
+        return currencyFilter.amountMicros.in.includes(value);
+      }
+      case currencyFilter.amountMicros.is !== undefined: {
+        if (currencyFilter.amountMicros.is === 'NULL') {
+          return value === null;
+        } else {
+          return value !== null;
+        }
+      }
+      default: {
+        throw new Error(
+          `Unexpected value for currency filter : ${JSON.stringify(
```

Comment:
Thank you for your contribution first of all I just had a small suggestion to change the keyword `value` to `operation` as it makes more sense. 

On top of the unit tests, I did some manual tests and it seems to be working. @charlesBochet

Reply:
Thanks for reviewing it @AbuHafsa
If we want to be accurate value should be amountMicros

Reply:
Thanks @AbuHafsa @charlesBochet for the suggestion, I will push the changes.